File: Easy/804. Unique Morse Code Words/uniqueMorseRepresentations.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        vector<string> str = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        set<string> res;
        //將每個word轉成摩斯密碼，再利用set來去重複，最後set裡面元素的各數就是答案
        for(int i = 0; i < words.size(); i++){
            string s = "";
            for(int j = 0; j < words[i].size(); j++){
                s += str[words[i][j] - 'a'];
            }
            res.insert(s);
        }
        return res.size();
    }
};

File: Easy/804. Unique Morse Code Words/uniqueMorseRepresentations.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        String [] s = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        Set res = new HashSet();
        for(int i = 0; i < words.length; i++){
            String sol = "";
            for(int j = 0; j < words[i].length(); j++){
                sol += s[words[i].charAt(j) - 'a'];
            }
            res.add(sol);
        }
        return res.size();
    }
}